src/lib.rs

```diff
@@ -47,115 +45,342 @@ pub enum Protection {
     ReadExecute,
 }
 
-impl Protection {
+// Anonymous mappings
 
-    fn as_open_options(self) -> fs::OpenOptions {
-        let mut options = fs::OpenOptions::new();
-        options.read(true)
-               .write(self == Protection::ReadWrite ||
-                      (self == Protection::ReadCopy && MmapInner::needs_write_for_copy()));
+/// Options that can be used to configure how an anonymous mapping is created.
+///
+/// Create this structure by calling [`memmap::anonymous()`](fn.anonymous.html),
+/// then chain call methods to configure additional options, finally, call [`map()`](#method.map)
+/// or [`map_mut()`](#method.map_mut).
+#[derive(Copy, Clone, Debug, Eq, PartialEq, Hash)]
+pub struct AnonymousMappingOptions {
+    protection: Option<Protection>,
+    len: usize,
+    stack: bool,
+}
+
+/// Configure a new anonymous mapping of `len` bytes.
+pub fn anonymous(len: usize) -> AnonymousMappingOptions {
+    AnonymousMappingOptions {
+        protection: None,
+        len,
+        stack: false,
+    }
+}
+
+impl AnonymousMappingOptions {
+    /// Make this mapping suitable to be a process or thread stack.
+    ///
+    /// This corresponds to `MAP_STACK` on Linux, which is currently a no-op.
+    pub fn stack(&mut self) -> &mut Self {
+        self.stack = true;
+        self
+    }
+
+    /// Set a protection to be used by this mapping.
+    pub fn protection(&mut self, protection: Protection) -> &mut Self {
+        self.protection = Some(protection);
+        self
+    }
+
+    fn map_inner(&self) -> Result<MmapInner> {
+        let inner = MmapInner::anonymous(self.len, self.protection.unwrap(), self.stack)?;
+        Ok(inner)
+    }
+
+    /// Actually map this anonymous mapping into the address space.
+    ///
+    /// This methos returns an immutable mapping, see [`map_mut()`](#method.map_mut)
+    /// for a mutable version.
+    ///
+    /// If the protection has not been [set explicitly](#method.protection), this method
+    /// assumes [`Read`](enum.Protection.html#variant.Read).
+    ///
+    /// # Errors
+    ///
+    /// This method returns `Err` when the underlying system call fails, which can happen for
+    /// a variety of reasons, such as when you don't have the necessary permissions for the file.
+    pub fn map(&self) -> Result<Mmap> {
```

Comment:
I think we should go ahead and remove it unless there's a compelling reason to keep it at this point.  We can always add it back later if a case can be made.